### PR TITLE
SC-715-use-unicode

### PR DESCRIPTION
**Pull request by potan**

### CODE REVIEW

**Review by msmolyakov — APPROVED** [review]

**Review by xrtm000 — CHANGES_REQUESTED** [review]
Add tests for all functions including empty string test, check that behavior of split function has not been changed, see SplitFunctionTest

File: lang/tests/src/test/scala/com/wavesplatform/lang/IntegrationTest.scala [hunk 1]
```diff
@@ -2174,4 +2174,88 @@ class IntegrationTest extends PropSpec with PropertyChecks with ScriptGen with M
...
+
+    val script5 = s"""dropRight("x冬x", 2)"""
+    eval(script5, version = ver) shouldBe
+      Right(CONST_STRING("x\ud87e").explicitGet())
+  }
+
+  property("unicode indexOf") {
+    val src =
+      """ "x冬xqweqwe".indexOf("we") """
+    genericEval[Environment, EVALUATED](src, ctxt = v5Ctx, version = V5, env = utils.environment) shouldBe Right(CONST_LONG(4L))
```
Comment (alexeykiselev) [inline lang/tests/src/test/scala/com/wavesplatform/lang/IntegrationTest.scala:2325]: What this function returns in V4 context? 
Also the funny script that may be tested
```
indexOf(take("世界x冬x", 4), take("冬", 1))
```
What it returns for V4 and V5?

Comment (potan) [reply lang/tests/src/test/scala/com/wavesplatform/lang/IntegrationTest.scala:2325]: All characters are 2-bytes. It returns 3 in both cases.

**Review by xrtm000 — APPROVED** [review]

**Review by mrkraft — APPROVED** [review]
